### PR TITLE
[WIP] SQLite promises

### PR DESCRIPTION
**Pull request by rnewman**
Early days. Trying to commit as I go.

### CODE REVIEW

File: app/services/sqlite.js [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+import sqlite3 from 'sqlite3';
+import thenifyAll from 'thenify-all';
+
+const Promise = global.Promise;
+
+export function verbose() {
+  sqlite3.verbose();
+}
+
+function thenifyMethods(source, dest, methods) {
```
Comment (ncalexan) [inline app/services/sqlite.js]: Worth a little comment.

File: app/services/storage.js [hunk 2]
```diff
@@ -18,28 +18,49 @@
...
+const mkdirp = thenify(cbmkdirp);
+
+export class Storage {
+  constructor(db) {
+    this.db = db;
+  }
+
+  /**
+   * Open a database and return a promise-based wrapper.
+   * @param path
```
Comment (ncalexan) [inline app/services/storage.js]: Update these comments, please.

File: app/services/storage.js [hunk 3]
```diff
@@ -18,28 +18,49 @@
  * @module Storage
  */
 
-import * as sqlite from './sqlite';
+import cbmkdirp from 'less-mkdirp';
```
Comment (ncalexan) [inline app/services/storage.js]: Did we already take this dep?  (I don't see it locally, and it's not in this patch.)

File: app/services/storage.js [hunk 4]
```diff
@@ -18,28 +18,49 @@
...
-        console.log(`Unable to fetch DB user_version: ${err}.`);
-      });
-  });
-
-  return o;
+import { DB, verbose } from './sqlite';
+
+const mkdirp = thenify(cbmkdirp);
+
+export class Storage {
```
Comment (ncalexan) [inline app/services/storage.js]: Class comment -- how is `Storage` different from ... Why is it not `ProfileStorage` or `EventStore` or ...

File: app/services/sqlite.js [hunk 5]
```diff
@@ -41,18 +41,19 @@ export class DB {
   /**
    *
    * @param filename the file path to open
-   * @param mode any mode flags to pass to `sqlite3.Database`.
+   * @param mode any mode flags to pass to `sqlite3.Database`. Defaults to OPEN_CREATE | OPEN_READWRITE.
    * @returns Promise that resolves to the opened database.
    */
   static open(filename, mode) {
```
Comment (ncalexan) [inline app/services/sqlite.js]: JS has default arguments now.

**Comment by ncalexan** [comment]
Couple nits and requests.  Might as well fold this into something more coherent before landing.

File: app/services/storage.js (hunk 3 above)
Comment (rnewman) [inline app/services/storage.js]: Yes: https://github.com/mozilla/tofino/pull/159/commits/994d93ae28b7686b7718490a0bfc69cb9e6570b4

File: app/services/storage.js (hunk 4 above)
Comment (rnewman) [inline app/services/storage.js]: I think this'll drop out later. It's really `ProfileStorage`, so I'll change the name.

**Pull request merged**